public/video-ui/src/components/Video/Display.js

```diff
@@ -62,15 +62,41 @@ class VideoDisplay extends React.Component {
     return this.props.video.expiryDate <= Date.now();
   };
 
-  renderEditButton = (editable, property) => {
+  isButtonDisabled = (property) => {
+
+    let formFields;
+    if (property === 'metadataEditable') {
+      formFields = ['title', 'category'];
+    } else if (property === 'youtubeEditable') {
+      formFields = ['youtubeCategory', 'youtubeChannel', 'privacyStatus'];
+    }
+    const errors  = validate(Object.assign(this.props.video, {
+      youtubeCategory: this.props.video.youtubeCategoryId,
+      youtubeChannel: this.props.video.channelId
+    }));
+
+    if (formFields.some(field => {
```

Comment:
if `property` != `'metadataEditable'` or  `'youtubeEditable'` it'll be `undefined` and this line will fail.

Could we instantiate `formFields` as `[]`? It also has the side-effect of becoming a `const` too.